### PR TITLE
fix issue #403: reset() func

### LINKED ISSUE #403: what does the reset() of MetricBucket do? seems no references
https://github.com/alibaba/sentinel-golang/blob/c362aaeafd1e500c9812d4961bcea81b25008b45/core/stat/base/metric_bucket.go#L68

another question:
the minrt updating function is not thread safe.
```go
func (mb *MetricBucket) AddRt(rt int64) {
	mb.addCount(base.MetricEventRt, rt)
	if rt < atomic.LoadInt64(&mb.minRt) { // here is not thread safe.
		// Might not be accurate here.
		atomic.StoreInt64(&mb.minRt, rt)
	}
}
```

### PR DESCRIPTION
**Pull request by Howie59**
### Does this pull request fix one issue?
Fixes #403

### CODE REVIEW

File: core/stat/base/bucket_leap_array.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ func (bla *BucketLeapArray) NewEmptyBucket() interface{} {
 }
 
 func (bla *BucketLeapArray) ResetBucketTo(bw *BucketWrap, startTime uint64) *BucketWrap {
+	b := bla.currentBucketWithTime(util.CurrentTimeMillis())
+	b.reset()
```
Comment (luckyxiaoqiang) [inline core/stat/base/bucket_leap_array.go:41]: What's the purpose of the added line? Please see the below line`bw.Value.Store(NewMetricBucket())`.

Comment (Howie59) [reply core/stat/base/bucket_leap_array.go:41]: for reset the old bucket and create new one

Comment (luckyxiaoqiang) [reply core/stat/base/bucket_leap_array.go:41]: In my view, `ResetBucketTo ` only intend to reset `bw`.

**Comment by sczyh30** [comment]
Any update on this PR?

**Pull request closed without merging**